src/firestore/firestore-internal.ts

```diff
@@ -25,14 +25,22 @@ import { App } from '../app';
 
 export class FirestoreService {
 
-  private appInternal: App;
-  private firestoreClient: Firestore;
+  private readonly appInternal: App;
+  private databases: Map<string, Firestore> = new Map();
 
   constructor(app: App) {
-    this.firestoreClient = initFirestore(app);
     this.appInternal = app;
   }
 
+  getDatabase(databaseId: string): Firestore {
+    let database = this.databases.get(databaseId);
+    if (database === undefined) {
```

Comment:
```suggestion
    if (typeof database === undefined) {
```

Reply:
Or we could also do `if (!this.databases.has(databaseId)) {` just a suggestion really. I will leave it up to you. :)